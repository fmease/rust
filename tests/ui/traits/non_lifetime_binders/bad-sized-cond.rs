// compile-flags: -Ztrait-solver=next

#![feature(non_lifetime_binders)]
//~^ WARN is incomplete and may not be safe

pub fn foo()
where
    for<V> V: Sized,
{
}

pub fn bar()
where
    for<V> V: IntoIterator,
{
}

fn main() {
    foo();

    bar();
    //~^ ERROR the trait bound `V: IntoIterator` is not satisfied
}
