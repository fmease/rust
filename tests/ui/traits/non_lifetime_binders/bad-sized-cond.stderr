warning: the feature `non_lifetime_binders` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/bad-sized-cond.rs:3:12
   |
LL | #![feature(non_lifetime_binders)]
   |            ^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #108185 <https://github.com/rust-lang/rust/issues/108185> for more information
   = note: `#[warn(incomplete_features)]` on by default

error[E0277]: the trait bound `V: IntoIterator` is not satisfied
  --> $DIR/bad-sized-cond.rs:21:5
   |
LL |     bar();
   |     ^^^ the trait `IntoIterator` is not implemented for `V`
   |
note: required by a bound in `bar`
  --> $DIR/bad-sized-cond.rs:14:15
   |
LL | pub fn bar()
   |        --- required by a bound in this function
LL | where
LL |     for<V> V: IntoIterator,
   |               ^^^^^^^^^^^^ required by this bound in `bar`

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0277`.
